src/systems/filecoin_mining/storage_mining/storage_miner_actor_state.go

```diff
@@ -176,9 +179,7 @@ func (st *StorageMinerActorState_I) _updateFailSector(rt Runtime, sectorNo secto
 	}
 
 	if newFaultCount > MAX_CONSECUTIVE_FAULTS {
-		// TODO: heavy penalization: slash pledge collateral and delete sector
-		// TODO: SendMessage(SPA.SlashPledgeCollateral)
-
+		// slashing is done at _slashCollateralForStorageFaults
```

Comment:
where is it called?

Reply:
this is now called in `_slashCollateralForStorageFaults` right after `_submitPowerReport`